### PR TITLE
DAM-LF minor fixes relative to GenMap

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR fixes some minor issue relative to GenMap.

In particular
  * validity of keys are checked when GenMap is created.
  * interface test
  * outdated comments

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala [hunk 1]
```diff
@@ -167,6 +167,7 @@ object SValue {
 
     def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
+      xs.foreach { case (k, v) => comparable(k) -> v }
```
Comment (SamirTalwar-DA) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: What does the `-> v` do?

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: you can take a look at `scala.Predef` 

```scala
 implicit final class ArrowAssoc[A](private val self: A) extends AnyVal {
    @inline def -> [B](y: B): (A, B) = (self, y)
    ...
  }
```

You problably already saw/used that when building map, for instance:

```scala
Map(1 -> "1", 2 -> "2", 3 -> "3)
```

or update a Map, for instance 

```scala
map + (2 -> "4")
```

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: Good catch. Thanks

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala [hunk 2]
```diff
@@ -167,6 +167,7 @@ object SValue {
 
     def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
+      xs.foreach { case (k, _) => comparable(k) }
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: Maybe we can avoid iterating the collection twice by side-effectful mapping with `breakOut`?

File: extractor/src/test/suite/scala/com/digitalasset/extractor/GenMapSpec.scala [hunk 3]
```diff
@@ -26,7 +26,7 @@ class GenMapSpec
 
   override protected def darFile =
     // FIXME https://github.com/digital-asset/daml/issues/2256
-    // Change to "daml-lf/encoder/test-1.8.dar" one 1.8 is frozen
+    // Change to "daml-lf/encoder/test-1.9.dar" one 1.9 is frozen
```
Comment (gerolf-da) [inline extractor/src/test/suite/scala/com/digitalasset/extractor/GenMapSpec.scala]:
```suggestion
    // Change to "daml-lf/encoder/test-1.9.dar" once 1.9 is frozen
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala (hunk 2 above)
Comment (S11001001) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: In the interests of forward-compatibility:

```suggestion
      xs.map { case p @ (k, _) => comparable(k); p }.to[O]
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala [hunk 4]
```diff
@@ -167,6 +167,7 @@ object SValue {
 
     def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
+      xs.map { case p @ (k, _) => comparable(k); p }.to[O]
       SGenMap(xs.to[O])
```
Comment (S11001001) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: Sorry, I should have been clearer; you can't use the `xs` variable twice like this.

```suggestion
      SGenMap(xs.map { case p @ (k, _) => comparable(k); p }.to[O])
```

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala]: You have been cleared. I was just planning to fix it latter. Thanks a lot anyway.

**Pull request merged**